persist-cli/src/main/java/io/ballerina/persist/cmd/Init.java

```diff
@@ -52,46 +60,67 @@ public class Init implements BLauncherCmd {
     private final PrintStream outStream = System.out;
     private final String configPath = PersistToolsConstants.CONFIG_SCRIPT_FILE;
 
-    private String name = "";
+    private String configName = "";
+    private String persistName = "";
     public String sourcePath = "";
     private static final String COMMAND_IDENTIFIER = "persist-init";
 
+
     Project balProject;
 
     @CommandLine.Option(names = {"-h", "--help"}, hidden = true)
     private boolean helpFlag;
 
     @Override
     public void execute() {
+        Path persistTomlPath = Paths.get(this.sourcePath, KEYWORD_PERSIST , PERSISTTOML);
+        Path databaseConfigPath = Paths.get(this.sourcePath, KEYWORD_MODULES, KEYWORD_CLIENTS,
+                DATABASE_CONFIGURATION_BAL);
         if (helpFlag) {
             String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(COMMAND_IDENTIFIER);
             errStream.println(commandUsageInfo);
             return;
         }
+        Generate persistCmd;
+        try {
+            Class<?> persistClass = Class.forName("io.ballerina.persist.cmd.Generate");
+            persistCmd = (Generate) persistClass.getDeclaredConstructor().newInstance();
+            persistCmd.setSourcePath(Paths.get(sourcePath).toAbsolutePath().toString());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
         try  {
             balProject = ProjectLoader.loadProject(Paths.get(sourcePath));
-            name = balProject.currentPackage().descriptor().org().value() + "." + balProject.currentPackage()
-                    .descriptor().name().value() + "." + "clients";
+            configName = balProject.currentPackage().descriptor().name().value() + "." + KEYWORD_CLIENTS;
+            persistName = balProject.currentPackage().descriptor().name().value() + "." + DATABASE;
         } catch (ProjectException e) {
             errStream.println("Not a Ballerina project (or any parent up to mount point)\n" +
                     "You should run this command inside a Ballerina project. ");
             return;
         }
+        File persistToml = new File(persistTomlPath.toString());
+        File databaseConfig = new File(databaseConfigPath.toString());
+        if (persistToml.exists() || databaseConfig.exists()) {
+            errStream.println("`bal persist init` command can only be used for initializing the project");
+            return;
+        }
         if (!Files.exists(Paths.get(sourcePath, configPath))) {
```

Comment:
Which file are we checking here? The logic seems to be the same for both if and else

Reply:
There seems to be confusion of logic here,
if Persist.toml exists
   // throw error
else
  // create Persist.toml
  // create database config bal
  if config.toml exists
     update file
  else
    create file
    
 By your logic following scenario are failures,
1. Only config.toml present -> Throws error need not be, user could have added the file never calling init
2. Line 107 is redundant, we know for ln 103 to pass, Config.toml should never be present

Reply:
databaseConfig file refers to `database_configuration.bal` not `Config.toml`

Reply:
Ack